client.go

```diff
@@ -46,6 +47,7 @@ func NewClient(conn net.Conn, cm *ConsumerManager) *Client {
 
 	return &Client{
 		id:          id,
+		monitor:     false,
```

Comment:
As a next step, each producer could have its own channel and flush on its connection on its own goroutine consuming from the global monitor channel. This does not block the current implementation though as it is just an implementation difference and most probably does not provide anything performance-wise.